db/migrate/20231027124052_add_placements_slug.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class AddPlacementsSlug < ActiveRecord::Migration[7.0]
+  disable_ddl_transaction!
+
+  def change
+    add_column :placements, :slug, :citext
+    add_index :placements, :slug, unique: true, algorithm: :concurrently
```

Comment:
Should it be more like 
```suggestion
    add_index :placements, [:slug, :trainee_id], unique: true, algorithm: :concurrently
```

Given that placement is only unique to the provider?

Reply:
It looks like slugs are just large random numbers so they aren't scoped as far as I can see. But it makes sense to do this I guess because we shouldn't really ever be looking up placements by slug alone.